src/App.vue

```
@@ -0,0 +1,251 @@
+      attemptedEmail.value = email;
+    }
+  } catch (error) {}
+};
+
+const handleRegister = async ({ email, name }) => {
+  try {
+    const newUser = await createUser({ email, name });
+    store.user = newUser;
+    needsRegistration.value = false;
+    await loadPosts();
+  } catch (error) {}
+};
+
+const handleLogout = () => {
+  store.user = null;
+  store.posts = [];
+  store.selectedPost = null;
+  store.isSidebarOpen = false;
+  needsRegistration.value = false;
+};
+
+const loadPosts = async () => {
+  store.arePostsLoading = true;
+  store.postsError = null;
+  try {
+    const allPosts = await getPosts();
+    store.posts = allPosts.filter((post) => post.userId === store.user.id);
+  } catch (error) {
+    store.postsError = 'Failed to load posts';
+  } finally {
+    store.arePostsLoading = false;
+  }
+};
+
+const handleCreateNewPost = () => {
+  store.selectedPost = null;
+  sidebarMode.value = 'create';
+  store.isSidebarOpen = true;
+  store.comments = [];
+};
+
+const handleSelectPost = async (post) => {
+  store.selectedPost = post;
+  sidebarMode.value = 'view';
+  store.isSidebarOpen = true;
+  await loadComments(post.id);
+};
+
+const handleSavePost = async (postData) => {
+  try {
+    if (sidebarMode.value === 'create') {
+      const newPost = await createPost({
+        ...postData,
+        userId: store.user.id,
+      });
+      store.posts.push(newPost);
+      store.selectedPost = newPost;
+      sidebarMode.value = 'view';
+      store.comments = [];
+    } else if (sidebarMode.value === 'edit') {
+      const updatedPost = await updatePost({
+        id: store.selectedPost.id,
+        ...postData,
+      });
+      const index = store.posts.findIndex((p) => p.id === updatedPost.id);
+      if (index !== -1) {
+        store.posts[index] = { ...store.posts[index], ...updatedPost };
+      }
+      store.selectedPost = { ...store.selectedPost, ...updatedPost };
+      sidebarMode.value = 'view';
+    }
+  } catch (error) {}
```

The task requires handling 'Add' and 'Update' errors. This `catch` block swallows any errors from `createPost` or `updatePost`. If saving the post fails, the user receives no notification, leading to a poor user experience and potential data loss. You should display an error message to the user.